pkg/service/backup/worker_scylla_upload.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2024 ScyllaDB
+
+package backup
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/scylladb/scylla-manager/v3/pkg/scyllaclient"
+	. "github.com/scylladb/scylla-manager/v3/pkg/service/backup/backupspec"
+	"github.com/scylladb/scylla-manager/v3/swagger/gen/scylla/v1/models"
+)
+
+// Decides whether we should use Scylla backup API for uploading the files.
+func (w *worker) useScyllaBackupAPI(ctx context.Context, d snapshotDir, hi hostInfo) (bool, error) {
+	// Scylla backup API does not handle creation of versioned files.
+	if d.willCreateVersioned {
+		return false, nil
```

Comment:
I miss more debug information like why ScyllaBackupAPI is not going to be used.
My proposal is to either replace (bool, error) returns with (error) return, where error!=nil means not supported. Error message can be logged (info) as a reasoning then.
I see you log which API is going to be used, so RClone api log info can be extended with reasoning,.

Reply:
I don't like mixing actual errors with checks, so I just added logging on failed checks.